saleor/order/models.py

```diff
@@ -132,14 +132,26 @@ def __str__(self):
     def get_absolute_url(self):
         return reverse('order:details', kwargs={'token': self.token})
 
+    def get_last_payment(self):
+        """Query-friendly method for getting last payment from cached
+        payments.
+        """
+        payments = list(self.payments.all())
+        if not payments:
+            return
+        payment_mapping = {payment.pk: payment for payment in payments}
+        latest_payment_pk = max(payment_mapping.keys())
+        last_payment = payment_mapping[latest_payment_pk]
+        return last_payment
```

Comment:
We have `PaymentQuerySet` which has overriden `last()` method - seems it's not used now. And I wonder if we could make this method easier, maybe this would work:
```payments = list(self.payments.all())
return payments[0] if payments else None```

Reply:
Or even:

```python
from operator import attrgetter

return max(self.payments.all(), key=attrgetter('pk'))
```